### PR TITLE
Importing rbd SDK modules to cephci

### PR DESCRIPTION
**Pull request by VasishtaShastry**
Signed-off-by: Vasishta <vashastr@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: cli/rbd/namespace.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        kw_copy = deepcopy(kw)
+        pool_spec = kw_copy.pop("pool_spec", "")
+        cmd_args = build_cmd_args(kw=kw_copy)
+
+        cmd = f"{self.base_cmd} create {pool_spec}{cmd_args}"
+
+        return self.execute(cmd=cmd)
+
+    def list_(self, **kw):
```
Comment (sunilangadi2) [inline cli/rbd/namespace.py:40]: I see some methods with _ at the end of the method name and some methods don't have,
Any reasons for such decisions?

Comment (VasishtaShastry) [reply cli/rbd/namespace.py:40]: @sunilangadi2 The methods which has _ are python keywords which cannot be used as name for methods.

**Review by sunilkumarn417 — COMMENTED** [review]
@VasishtaShastry , Thanks.
can we execute RBD suites to ensure consistency?

File: cli/rbd/client.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from copy import deepcopy
+
+from cli.utilities.utils import build_cmd_args
+
+
+class Client:
```
Comment (vamahaja) [inline cli/rbd/client.py:6]: To use `self.execute`, class needs to be inherited from `Cli`.
Refer - https://github.com/red-hat-storage/cephci/blob/master/cli/__init__.py

File: cli/rbd/client.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                client-id(str): client ID (leave unspecified to disconnect all)
+        Returns:
+            Dict(str):
+            A mapping of host strings to the given task's return value for that host's execution run.
+        """
+        kw = kw.get("kw")
+
+        kw_copy = deepcopy(kw)
+        journal_spec = kw_copy.pop("journal_spec", "")
+        cmd_args = build_cmd_args(kw=kw_copy)
```
Comment (vamahaja) [inline cli/rbd/client.py:38]: `build_cmd_args` function has different name, need to rename - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py#L149

File: cli/rbd/client.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        cmd = f"{self.base_cmd} disconnect {journal_spec}{cmd_args}"
+
+        return self.execute(cmd=cmd)
```
Comment (vamahaja) [inline cli/rbd/client.py:42]: Don't it need to be executed with `root` privilege ?

File: cli/rbd/config/config_global.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+              key(str): This is the config key.
+
+        Returns:
+          Dict(str)
+            A mapping of host strings to the given task’s return value for that host’s execution run
+        """
+        kw = kw.get("kw")
+        config_entity = kw.get("config_entity")
+        key = kw.get("key")
+        cmd = self.base_cmd + f" get {config_entity} {key}"
```
Comment (vamahaja) [inline cli/rbd/config/config_global.py:31]: Can't it be modified as -
```
cmd = f"{self.base_cmd} get {config_entity} {key}"
```

File: cli/rbd/config/config_global.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+class Global:
+    """
+    This module provides CLI interface to manage the configuration override.
+    """
+
+    def __init__(self, nodes, base_cmd):
+        self.base_cmd = base_cmd + " global"
+
+    def get_(self, **kw):
```
Comment (vamahaja) [inline cli/rbd/config/config_global.py:14]: Why `_` at the end of each function ?

File: cli/rbd/config/config_global.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            A mapping of host strings to the given task’s return value for that host’s execution run
+        """
+        kw = kw.get("kw")
+        config_entity = kw.get("config_entity")
+        key = kw.get("key")
+        cmd = self.base_cmd + f" get {config_entity} {key}"
+
+        return self.execute(cmd=cmd)
+
+    def list_(self, **kw):
```
Comment (vamahaja) [inline cli/rbd/config/config_global.py:35]: We decided to pass mandatory parameters as method signature and optional parameters through `kw`

**Review by vamahaja — CHANGES_REQUESTED** [review]
Please raise multiple PR for classes. It will be easy to review

**Pull request closed without merging**